solidity/contracts/isms/ccip-read/AbstractCcipReadIsm.sol

```diff
@@ -21,10 +21,50 @@ import {PackageVersioned} from "../../PackageVersioned.sol";
  *    message. This functions selector should be provided as the `callbackFunction` payload
  *    for the OffchainLookup error
  */
-abstract contract AbstractCcipReadIsm is ICcipReadIsm, PackageVersioned {
+abstract contract AbstractCcipReadIsm is ICcipReadIsm, MailboxClient {
     // ============ Constants ============
 
     // solhint-disable-next-line const-name-snakecase
     uint8 public constant moduleType =
         uint8(IInterchainSecurityModule.Types.CCIP_READ);
+
+    string[] internal _urls;
+
+    /**
+     * @notice Emitted when new CCIP-read urls are being set
+     */
+    event UrlsChanged(string[] newUrls);
+
+    function setUrls(string[] memory __urls) public onlyOwner {
```

Comment:
If we do generalize this, I wonder if we should make urls append-only. Imo that is a better story to avoid owners being able to rug by removing urls?